src/api/providers/bedrock.ts

```diff
@@ -396,9 +420,26 @@ Please check:
 						errorStack: error.stack,
 						errorMessage: error.message,
 					})
-					yield {
-						type: "text",
-						text: `Error: ${error.message}`,
+
+					// Check if error is related to credentials and we're in a remote SSH context
+					const errorMessage = error.message.toLowerCase()
+					if (
+						(errorMessage.includes("credentials") ||
+							errorMessage.includes("profile") ||
+							errorMessage.includes("not found") ||
+							errorMessage.includes("cannot load credentials")) &&
+						this.options.awsUseProfile &&
+						isRemoteSSH()
+					) {
```

Comment:
Nitpick, but could you put this into a `isErrorRelatedToCredentials(errorMessage)` method that you can reuse below?